### ISSUE TITLE
%eyre garbage disposal: properly cancel incoming requests

### ISSUE THREAD

**Issue by ohAitch** [urbit/urbit#368]
Store outgoing request types by duct, and upon disappearance:
- cancel ford requests
- delete active long-polls: wasp, ixor (allowing the printf for #361 to be restored; trigger #367 cleanup)
- propagate remote cancel
- authenticate remote cancel

**Comment by ohAitch** [comment]
not doing this slows down clay sync, because dependency polls to old revisions never get cleaned up.

**Comment by philipcmonk** [comment]
As overheard on the interwebz (posting here for easy linkage):

So, I think I've found the issue here, and Anton and I have come up with a workaround.  Firstly, several hundred revisions of a random markdown file caused no slowdown.  Secondly, a hundred revisions of adding comments to ape/talk.hoon caused no slowdown.  Thirdly, a hundred revisions of adding "." lines to the end of ape/talk.hoon caused no slowdown.

I did reproduce the problem by adding a "." to the end of pub/talk/fab/hymn.hook while having talk open.  What we believe is that every time we autoreload talk, we generate a new request to urb.js, which causes eyre to ask ford to ask clay to watch all of its dependencies.  All fine and good.  When an update happens again, the dependencies for talk have changed, so we autoreload, and eyre tells ford to tell clay to not listen for more updates on the old set of dependencies.  It then says to listen for updates on the new set of dependencies and sends the new page to the browser.  The browser asks again for urb.js, but this is for urb.js on the newest revision, which may have a different dependency set.  Thus, the old urb.js request's dependencies are never canceled, which means that clay has to continually check if they change.  What should happen, but doesn't, is that whenever an http request is canceled, we cancel its subscriptions.  The workaround is to somehow kick urb.js's dependencies.

tl;dr is that eyre doesn't properly dispose of dead connections, as described in https://github.com/urbit/urbit/issues/368 .  The temporary workaround is to make a trivial change to a mark whenever things get annoyingly slow.  This can be just `echo "::" >> pier/home/mar/atom.hoon`.
